handlers/meshery_pattern_handler.go

```diff
@@ -181,6 +183,56 @@ func (h *Handler) handlePatternPOST(
 		formatPatternOutput(rw, byt, format)
 		return
 	}
+
+	if parsedBody.K8sManifest != "" {
+		pattern, err := pCore.NewPatternFileFromK8sManifest(parsedBody.K8sManifest, false)
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("failed to convert to pattern: %s", err), http.StatusBadRequest)
+			return
+		}
+
+		patternYAML, err := pattern.ToYAML()
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("failed to generate pattern: %s", err), http.StatusInternalServerError)
+			return
+		}
+
+		name, err := models.GetPatternName(string(patternYAML))
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("failed to get pattern name: %s", err), http.StatusInternalServerError)
+			return
+		}
+
+		patternModel := &models.MesheryPattern{
+			Name:        name,
```

Comment:
We might end up with to other attributes:

CANONICAL: "MESHERY001"
VERSION: "v0.0.0"